src/Form/CropTypeForm.php

```diff
@@ -71,14 +78,21 @@ protected function actions(array $form, FormStateInterface $form_state) {
   /**
    * {@inheritdoc}
    */
-  public function validate(array $form, FormStateInterface $form_state) {
-    parent::validate($form, $form_state);
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
 
     $id = trim($form_state->getValue('id'));
+    $aspect_ratio = $form_state->getValue('aspect_ratio');
+
     // '0' is invalid, since elsewhere we check it using empty().
     if ($id == '0') {
       $form_state->setErrorByName('type', $this->t("Invalid machine-readable name. Enter a name other than %invalid.", array('%invalid' => $id)));
     }
+
+    // If aspect ratio is set verify format.
+    if (!empty($aspect_ratio) && !preg_match("#^[0-9:]+$#", $aspect_ratio)) {
+      $form_state->setErrorByName('aspect_ratio', $this->t("Invalid format of aspect ratio. Enter a ratio in the format %format.", array('%format' => 'W:H')));
```

Comment:
Yes i think only in this case (if user have set an aspect ratio) we need to verify if this format is really an ratio type W:H to use it in UI. What would be your idea about this?